src/facade/dragonfly_listener.h

```diff
@@ -25,7 +25,8 @@ class ServiceInterface;
 
 class Listener : public util::ListenerInterface {
  public:
-  Listener(Protocol protocol, ServiceInterface*);
+  enum class Role { PRIVILEGED, MAIN, OTHER };
+  Listener(Protocol protocol, ServiceInterface*, Role role = Role::OTHER);
```

Comment:
It would be helpful to have a comment here explaining what those enum values are actually are, so readers don't have to look at the code to undertstand

Reply:
Also I didn't see any uses of Role::OTHER 🤔